src/containers/DAO/pages/DAOPage/Proposals/CreateProposal/hooks/useCreateProposal.ts

```diff
@@ -0,0 +1,89 @@
+import { useState, useMemo, useCallback } from 'react';
+import { useHistory } from 'react-router-dom';
+import { useWeb3React } from '@web3-react/core';
+import { Web3Provider } from '@ethersproject/providers/lib/web3-provider';
+import type { zDAO } from '@zero-tech/zdao-sdk';
+import { useDidMount } from 'lib/hooks/useDidMount';
+import { useWillUnmount } from 'lib/hooks/useWillUnmount';
+import { getTokenOptionsFromAssets } from '../CreateProposal.helpers';
+import { DAO_CREATE_PROPPAL } from '../../../Proposals/Proposals.constants';
+import useAssets from '../../../hooks/useAssets';
+
+export const useCreateProposal = (dao?: zDAO) => {
```

Comment:
I think this hook is doing a lot of stuff it shouldn't be responsible for, such as handling navigation, component lifecycle, and "Connect Wallet" modal state.